storage/rocksdb/rdb_datadic.cc

```diff
@@ -2964,34 +2967,56 @@ bool Rdb_key_def::is_variable_length_field(const enum_field_types type) {
   return type == MYSQL_TYPE_VARCHAR || is_blob(type);
 }
 
+template <const int bytes>
+bool check_src_len(uint src_len);
+
+template <>
+bool check_src_len<2>(uint src_len) {
+  if (src_len & 1) {
+    /*
+      utf8mb3 characters are encoded into two-byte entities. There is no way
+      we can have an odd number of bytes after encoding.
+    */
+    return false;
+  }
+  return true;
+}
+
+template <>
+bool check_src_len<3>(uint src_len) {
+  if (src_len % 3) {
+    /*
+      utf8mb4 characters are encoded into three-byte entities. There is no way
+      we can have 1 or 2 bytes after encoding.
+    */
+    return false;
+  }
+  return true;
+}
+
 /*
   Unpack data that has charset information.  Each two bytes of the input is
   treated as a wide-character and converted to its multibyte equivalent in
   the output.
  */
-static int unpack_charset(
-    const CHARSET_INFO *cset,  // character set information
-    const uchar *src,          // source data to unpack
-    uint src_len,              // length of source data
-    uchar *dst,                // destination of unpacked data
-    uint dst_len,              // length of destination data
-    uint *used_bytes)          // output number of bytes used
+template <const int bytes>
+int unpack_charset(const CHARSET_INFO *cset,  // character set information
```

Comment:
It seems like this template is unused at the moment since it's meant for no_pad 2/3 byte characters. Was the intention to keep unpack_binary_varlength as a standalone function, rather than templatize it like unpack_binary_or_utf8_varlength_space_pad? Is this function being kept to support possibly utf8mb4_0900_bin collations in the future?

Reply:
I didn't notice that `unpack_charset` is now unused. I removed it.